src/main/java/net/dv8tion/jda/core/handle/MessageReactionBulkRemoveHandler.java

```diff
@@ -54,10 +60,33 @@ protected Long handleInternally(JSONObject content)
             EventCache.LOG.debug("Received a reaction for a channel that JDA does not currently have cached");
             return null;
         }
-        api.getEventManager().handle(
-                new MessageReactionRemoveAllEvent(
-                        api, responseNumber,
-                        messageIdString, channel));
+        IEventManager manager = api.getEventManager();
+
+        switch (channel.getType())
+        {
+            case TEXT:
+               manager.handle(
+                   new GuildMessageReactionRemoveAllEvent(
+                           api, responseNumber,
+                           messageId, (TextChannel) channel));
+               break;
+            case GROUP:
+                manager.handle(
+                    new GroupMessageReactionRemoveAllEvent(
+                            api, responseNumber,
+                            messageId, (Group) channel));
+                break;
+            case PRIVATE:
+                manager.handle(
+                    new PrivateMessageReactionRemoveAllEvent(
+                            api, responseNumber,
+                            messageId, (PrivateChannel) channel));
+        }
+
+        manager.handle(
+            new MessageReactionRemoveAllEvent(
+                    api, responseNumber,
+                    messageId, channel));
```

Comment:
No. I do not want this changed.
The idea is that there are combined events and specific events. MessageReaction* is similar to the combined events for normal messages being stuff like MessageReceivedEvent.
To make them extend would be to remove the fact that they are considered "Combined" events.